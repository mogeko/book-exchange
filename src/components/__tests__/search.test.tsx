import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Search from "@/components/search";

describe("Search", () => {
  it("Create a search bar", () => {
    const { container } = render(<Search />);
    expect(container).toBeInTheDocument();
    expect(container.querySelector("input")).toBeInTheDocument();
  });

  it("snapshot a search bar", () => {
    const { container } = render(<Search />);
    expect(container).toMatchSnapshot();
  });
});
