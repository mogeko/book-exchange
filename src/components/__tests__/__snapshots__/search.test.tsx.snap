// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot a search bar 1`] = `
<div>
  <div
    class="hidden md:form-control"
  >
    <input
      class="input input-bordered input-primary input-sm w-96"
      placeholder="Search"
      type="text"
    />
  </div>
  <a
    class="btn btn-ghost btn-circle md:hidden"
    href="/#"
  >
    <svg
      aria-hidden="true"
      class="h-6 w-6"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
        fill-rule="evenodd"
      />
    </svg>
  </a>
</div>
`;
